src/components/refactor/dialogs/load-modification/load-modification-form.js

```diff
@@ -0,0 +1,166 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import TextInput from '../../rhf-inputs/text-input';
+import {
+    ACTIVE_POWER,
+    EQUIPMENT_ID,
+    EQUIPMENT_NAME,
+    LOAD_TYPE,
+    REACTIVE_POWER,
+} from '../../utils/field-constants';
+import {
+    ActivePowerAdornment,
+    filledTextField,
+    gridItem,
+    GridSection,
+    ReactivePowerAdornment,
+} from '../../../dialogs/dialogUtils';
+import SelectInput from '../../rhf-inputs/select-input';
+import { getLoadTypeLabel, LOAD_TYPES } from '../../../network/constants';
+import FloatInput from '../../rhf-inputs/float-input';
+import Grid from '@mui/material/Grid';
+import { useCallback, useEffect, useState } from 'react';
+import {
+    fetchEquipmentInfos,
+    fetchEquipmentsIds,
+} from '../../../../utils/rest-api';
+import AutocompleteInput from '../../rhf-inputs/autocomplete-input';
+import { useWatch } from 'react-hook-form';
+import { useIntl } from 'react-intl';
+
+const LoadModificationForm = ({ currentNode, studyUuid }) => {
+    const currentNodeUuid = currentNode?.id;
+    const [equipmentOptions, setEquipmentOptions] = useState([]);
+    const [loadingEquipmentOptions, setLoadingEquipmentOptions] =
+        useState(true);
+    const [loadInfos, setLoadInfos] = useState(null);
+    const intl = useIntl();
+
+    const loadId = useWatch({
+        name: `${EQUIPMENT_ID}`,
+    });
+
+    useEffect(() => {
+        fetchEquipmentsIds(
+            studyUuid,
+            currentNodeUuid,
+            undefined,
+            'LOAD',
+            true
+        ).then((values) => {
+            setEquipmentOptions(values);
+            setLoadingEquipmentOptions(false);
+        });
+    }, [studyUuid, currentNodeUuid]);
+
+    useEffect(() => {
+        if (loadId) {
+            fetchEquipmentInfos(
+                studyUuid,
+                currentNodeUuid,
+                'loads',
+                loadId,
+                true
+            ).then((value) => {
+                if (value) {
+                    setLoadInfos(value);
+                }
+            });
+        } else {
+            setLoadInfos(null);
+        }
+    }, [studyUuid, currentNodeUuid, loadId]);
+
+    const getObjectId = useCallback((object) => {
+        if (typeof object === 'string') {
+            return object;
+        }
+
+        return object.id;
+    }, []);
+
+    const loadIdField = (
+        <AutocompleteInput
+            allowNewValue={true}
+            forcePopupIcon
+            //hack to work with freesolo autocomplete
+            //setting null programatically when freesolo is enable wont empty the field
+            name={`${EQUIPMENT_ID}`}
+            label="ID"
+            options={
+                loadingEquipmentOptions
+                    ? []
+                    : equipmentOptions.sort((a, b) => a.localeCompare(b))
```

Comment:
```suggestion
                equipmentOptions.sort((a, b) => a.localeCompare(b))
```

Maybe there is a behaviour I can't see here, but I don't understand this boolean "loadingEquipmentOptions"